Short-Answer/Algorithms_Answers.md

```diff
@@ -3,13 +3,26 @@
 ## Exercise I
 
 a)
-
+I think, the overall runtime complexity for the 1st problem is O(n). There is only one loop involved in this snippet and the runtime will change linearly with n.
 
 b)
+My guess is, the runtime complexity would be O(n^n) for the problem-2 since this snippet has nested loops('while' and 'for' loop).
```

Comment:
There are nested loops, so that's a good catch. However, the total time complexity isn't `O(n^n)`.

The outer loop has a complexity of `O(n)` because we increment 1 every time. The inner loop however has a time complexity of `O(logn)` because we're doubling j every loop. Instead of moving one number at a time, j is getting bigger and bigger until the loop ends, so it's faster than just `O(n)`

Because the loops are nested, we multiply the nested time complexities. Once you multiply both time complexities together, the final result is `O(nlogn)`.

Reply:
What a great explanation! Thanks so much for clarifying :-)